src/transformStateWithClones.js

```diff
@@ -1,13 +1,48 @@
 'use strict';
-
 /**
  * @param {Object} state
  * @param {Object[]} actions
- *
- * @return {Object[]}
+ * @returns {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const nextState = { ...state };
+
+  for (const currentAction of actions) {
+    switch (currentAction.type) {
+      case 'addProperties':
+        const extraData = currentAction.extraData;
```

Comment:
You can use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)